controllers/nonadmin_controller.go

```diff
@@ -166,7 +174,7 @@ func ensureRequiredSpecs(deploymentObject *appsv1.Deployment, image string, imag
 			Name:            nonAdminObjectName,
 			Image:           image,
 			ImagePullPolicy: imagePullPolicy,
-			Env:             []corev1.EnvVar{namespaceEnvVar},
+			Env:             []corev1.EnvVar{namespaceEnvVar, enforceBackupSpecEnvVar},
```

Comment:
So, goal here is admin user will enforce some NonAdminBackup spec field values through DPA (for example, `SnapshotVolumes` must be `false` to every NonAdminBackup created)

During NAC startup, it reads these values admin user has set. 

we could read from DPA directly, doing a DPA list call in OADP namespace before here https://github.com/migtools/oadp-non-admin/blob/8d8de636b68c0241b937403a98ab4384616458ba/cmd/main.go#L135 and passing result to `NonAdminBackupReconciler`. Problem with this approach are:
1. need to change NAC RBAC (but easy to fix, just add to NAC role list DPA)
2. how to tell NAC deployment to restart when that field changes? 

Current implementation idea is passing DPA field value as env var to deployment. On NAC controller, it would Unmarshal the value and use it. Problem with this approach is that we can reach a char limit on env var value.

Any ideas on a third solution? Or how to fix problem `2` of first idea?

Reply:
> how to tell NAC deployment to restart when that field changes?

if NAC is in a deployment, simply os.Exit() should result in another pod coming up.

Reply:
changed logic here

now I save DPA `spec.nonAdmin.enforceBackupSpec` that created Deployment and compare with current DPA `spec.nonAdmin.enforceBackupSpec`, only update Deployment when those are different